CompilerSource/compiler/components/write_object_data.cpp

```diff
@@ -657,12 +656,10 @@ static inline void write_object_data_structs(std::ostream &wto,
         << resname(object->parent_name)  << ", "
         << object->id
         << " },\n";
-    if (object->id >= obmx) obmx = object->id;
   }
   wto.unsetf(ios_base::floatfield);
   wto << "  };\n";
   wto << "  int objectcount = " << parsed_objects.size() << ";\n";
-  wto << "  int obj_idmax = " << obmx+1 << ";\n";
```

Comment:
Hold on, there, slick. This is one version. Where's the other? Is it computed at load time? It's better to compute things at compile time (so they'll be available no matter whose code runs first).

Reply:
The other version, `object_idmax`, is in `IDE_EDIT_resourcenames.h` and you are computing it at compile time generically for each group of resources.
https://github.com/enigma-dev/enigma-dev/blob/f9fde996b4831a9e6798f19d85b4da689e7359e1/CompilerSource/compiler/compile.cpp#L177